lib/src/warnings.dart

```diff
@@ -455,14 +464,21 @@ class PackageWarningOptions {
     return newOptions;
   }
 
+  void error(PackageWarning kind) =>
+      warningModes[kind] = PackageWarningMode.error;
+
   void ignore(PackageWarning kind) =>
       warningModes[kind] = PackageWarningMode.ignore;
 
   void warn(PackageWarning kind) =>
       warningModes[kind] = PackageWarningMode.warn;
 
-  void error(PackageWarning kind) =>
-      warningModes[kind] = PackageWarningMode.error;
+  void writeErrorOnDeprecation(PackageWarning warning) {
+    if (warning._isDeprecated) {
+      stderr.writeln("The warning 'missingCodeBlockLanguage' is deprecated. "
```

Comment:
I guess this should refer to `_flagName`. And then we don't need the second sentence; it can just be spelled out in the help text.